### PR TITLE
Lnk 1878 dead letter workflow census

### PR DESCRIPTION
**Pull request by edward-miller-lcg**

### CODE REVIEW

File: Census/Listeners/CensusListener.cs [hunk 1]
```diff
@@ -57,67 +59,96 @@ private async System.Threading.Tasks.Task StartConsumerLoop(CancellationToken ca
...
+                    rawmessage = kafkaConsumer.Consume(cancellationToken);
                 }
-                catch (Exception ex)
+                catch(ConsumeException ex)
                 {
-                    var errorMessage = $"Unable to deserialize message: {rawmessage?.Message?.Value}";
-                    _logger.LogError(errorMessage);
+                    _nonTransientExceptionHandler.HandleException(ex);
+                    _logger.LogError($"Error consuming message: {ex.Error.Reason}");
                     continue;
```
Comment (dvargaslantana) [inline Census/Listeners/CensusListener.cs:74]: Hey Ed, the event needs to be committed here after the nontransient handles the exception

File: Census/Listeners/CensusListener.cs [hunk 2]
```diff
@@ -57,67 +59,96 @@ private async System.Threading.Tasks.Task StartConsumerLoop(CancellationToken ca
...
+                    IBaseMessage message = null;
+                    try
+                    {
+                        message = DeserializeMessage(rawmessage.Topic, rawmessage.Message.Value);
+                    }
+                    catch (Exception ex)
+                    {
+                        _nonTransientExceptionHandler.HandleException(rawmessage,ex);
+                        var errorMessage = $"Unable to deserialize message: {rawmessage?.Message?.Value}";
+                        _logger.LogError(errorMessage);
```
Comment (dvargaslantana) [inline Census/Listeners/CensusListener.cs:91]: Hey Ed, the event needs to be committed here after the nontransient handles the exception

File: Census/Listeners/CensusListener.cs [hunk 3]
```diff
@@ -57,67 +59,96 @@ private async System.Threading.Tasks.Task StartConsumerLoop(CancellationToken ca
...
-                        _ => null
-                    };
+                            throw new MissingFacilityIdException("No Facility ID provided. Unable to process message.");
+                        }
+                    }
+                    catch (MissingFacilityIdException ex)
+                    {
+                        _nonTransientExceptionHandler.HandleException(rawmessage, ex);
+                        var errorMessage = $"No Facility ID provided. Unable to process message: {message}";
+                        _logger.LogWarning(errorMessage);
```
Comment (dvargaslantana) [inline Census/Listeners/CensusListener.cs:110]: Hey Ed, the event needs to be committed here after the nontransient handles the exception

File: Census/Listeners/CensusListener.cs (hunk 1 above)
Comment (edward-miller-lcg) [reply Census/Listeners/CensusListener.cs:75]: added

File: Census/Listeners/CensusListener.cs (hunk 2 above)
Comment (edward-miller-lcg) [reply Census/Listeners/CensusListener.cs:93]: added

File: Census/Listeners/CensusListener.cs (hunk 3 above)
Comment (edward-miller-lcg) [reply Census/Listeners/CensusListener.cs:113]: added

**Review by dvargaslantana — APPROVED** [review]

**Pull request merged**